src-java/testing/functional-tests/src/main/groovy/org/openkilda/functionaltests/helpers/SwitchHelper.groovy

```diff
@@ -84,49 +106,53 @@ class SwitchHelper {
                     MULTITABLE_POST_INGRESS_DROP_COOKIE, MULTITABLE_EGRESS_PASS_THROUGH_COOKIE,
-                    MULTITABLE_TRANSIT_DROP_COOKIE, LLDP_POST_INGRESS_COOKIE, LLDP_POST_INGRESS_ONE_SWITCH_COOKIE]
-            if (sw.features.contains(SwitchFeature.NOVIFLOW_COPY_FIELD)) {
-                multiTableRules.add(LLDP_POST_INGRESS_VXLAN_COOKIE)
+                    MULTITABLE_TRANSIT_DROP_COOKIE, LLDP_POST_INGRESS_COOKIE, LLDP_POST_INGRESS_ONE_SWITCH_COOKIE,
+                    ARP_POST_INGRESS_COOKIE, ARP_POST_INGRESS_ONE_SWITCH_COOKIE]
+            if (sw.features.contains(SwitchFeature.NOVIFLOW_PUSH_POP_VXLAN)) {
+                multiTableRules.addAll([LLDP_POST_INGRESS_VXLAN_COOKIE, ARP_POST_INGRESS_VXLAN_COOKIE])
             }
             northbound.getLinks(sw.dpId, null, null, null).each {
                 if (sw.features.contains(SwitchFeature.NOVIFLOW_COPY_FIELD)) {
-                    multiTableRules.add(Cookie.encodeIslVxlanEgress(it.source.portNo))
-                    multiTableRules.add(Cookie.encodeIslVxlanTransit(it.source.portNo))
+                    multiTableRules.add(encodeIslVxlanEgress(it.source.portNo))
+                    multiTableRules.add(encodeIslVxlanTransit(it.source.portNo))
                 }
-                multiTableRules.add(Cookie.encodeIslVlanEgress(it.source.portNo))
+                multiTableRules.add(encodeIslVlanEgress(it.source.portNo))
             }
             northbound.getSwitchFlows(sw.dpId).each {
                 if (it.source.datapath.equals(sw.dpId)) {
-                    multiTableRules.add(Cookie.encodeIngressRulePassThrough(it.source.portId))
+                    multiTableRules.add(encodeIngressRulePassThrough(it.source.portId))
                     if (swProps.switchLldp || it.source.detectConnectedDevices.lldp) {
-                        switchLldpRules.add(Cookie.encodeLldpInputCustomer(it.source.portId))
+                        switchLldpRules.add(encodeLldpInputCustomer(it.source.portId))
                     }
                 }
                 if (it.destination.datapath.equals(sw.dpId)) {
-                    multiTableRules.add(Cookie.encodeIngressRulePassThrough(it.destination.portId))
+                    multiTableRules.add(encodeIngressRulePassThrough(it.destination.portId))
                     if (swProps.switchLldp || it.destination.detectConnectedDevices.lldp) {
-                        switchLldpRules.add(Cookie.encodeLldpInputCustomer(it.destination.portId))
+                        switchLldpRules.add(encodeLldpInputCustomer(it.destination.portId))
                     }
                 }
             }
         }
         if (swProps.switchLldp) {
             switchLldpRules.addAll([LLDP_INPUT_PRE_DROP_COOKIE, LLDP_TRANSIT_COOKIE, LLDP_INGRESS_COOKIE])
         }
+        if (swProps.switchArp) {
+            switchLldpRules.addAll([ARP_INPUT_PRE_DROP_COOKIE, ARP_TRANSIT_COOKIE, ARP_INGRESS_COOKIE])
```

Comment:
nit: not quite accurate name `switchLldpRules` because we add ARP rules to this list

Reply:
fixed